### ISSUE TITLE
Macro expansion producing 'defined' has undefined behavior

### ISSUE THREAD

**Issue by johntmcintosh** [TextureGroup/Texture#872]
After updating to Xcode 9.3, when building on master, I'm seeing several new build warnings related to:

```
macro expansion producing 'defined' has undefined behavior
```

A bit of [stackoverflow searching](https://stackoverflow.com/questions/42074035/how-to-deal-with-clangs-3-9-wexpansion-to-defined-warning?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa) seems to point to the fact that the warning is triggered by using `defined` outside of being inline with a `#if`. For example, this would throw the warning:

```
#define AS_KDEBUG_ENABLE defined(PROFILE) && __has_include(<sys/kdebug_signpost.h>)
#if AS_KDEBUG_ENABLE
```

But this would be ok:

```
#if defined(PROFILE) && __has_include(<sys/kdebug_signpost.h>)
```

However, because in this example `AS_KDEBUG_ENABLE` needs to be referenced from other `#if`s as well, the solution is not as simple as just inlining it like above.

I'm not comfortable enough with what the macros here are doing to submit a PR with a proposed solution, but wanted to go ahead and document what's going on so that someone else hopefully can.

![screen shot 2018-04-05 at 10 29 40 am](https://user-images.githubusercontent.com/931283/38375630-474aeea0-38bc-11e8-983e-34d516dfc413.png)

**Comment by andrewrohn** [comment]
Dealing with same issue here. Warning is preventing us from building. :/ Looking into it soon.

**Comment by maicki** [comment]
Fixed via #879
